pkg/tfvars/vsphere/vsphere.go

```diff
@@ -130,3 +145,110 @@ func convertVCentersToMap(values []vtypes.VCenter) map[string]vtypes.VCenter {
 	}
 	return vcenterMap
 }
+
+func getSubnetMask(prefix netip.Prefix) (string, error) {
+	prefixLength := net.IPv4len * 8
+	if prefix.Addr().Is6() {
+		prefixLength = net.IPv6len * 8
+	}
+	ipMask := net.CIDRMask(prefix.Masked().Bits(), prefixLength)
+	maskBytes, err := hex.DecodeString(ipMask.String())
+	if err != nil {
+		return "", err
+	}
+	ip := net.IP(maskBytes)
+	maskStr := ip.To16().String()
+	return maskStr, nil
+}
+
+func constructKargsFromNetworkConfig(ipAddrs []string, nameservers []string, gateway string) (string, error) {
+	if logrus.IsLevelEnabled(logrus.TraceLevel) {
+		logrus.Tracef("Constructing kargs from IPs [%v] nameservers [%v] gateway [%v]", ipAddrs, nameservers, gateway)
+	}
+	outKargs := ""
+
+	var gatewayIP netip.Addr
+	if len(gateway) > 0 {
+		ip, err := netip.ParseAddr(gateway)
+		if err != nil {
+			return "", err
+		}
+		if ip.Is6() {
+			gateway = fmt.Sprintf("[%s]", gateway)
+		}
+		gatewayIP = ip
+	}
+
```

Comment:
Can there be a mixture of IPv4 and IPv6 addresses? Should the gateway, nameserver and IP addresses if present should all be of the same IP address family?